### PR TITLE
Mvtoptions

### PR DESCRIPTION
**Pull request by abenrob**
possible solution for #22 
relying on the arguments object of Tilesplash.prototype.layer, but not at all sure if this is affecting functionality that I am not using, or is underdocumented. @devTristan Please review and suggest changes.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -185,10 +186,28 @@ function stringifyTopojson(layers){
 
 Tilesplash.prototype.layer = function(name){
   var callbacks = Array.prototype.slice.call(arguments);
+
+  // shave off name
   callbacks.shift();
+
+  var mvtOptions = {};
+  // if options were passed, we'll have 2 left
+  if (callbacks.length === 2){
```
Comment (abenrob) [inline lib/index.js]: Just trying to get access to the passed in mvtOptions. Using `argumants` makes this tricky, and I'm unsure on the best way to do it.

File: lib/index.js [hunk 2]
```diff
@@ -28,11 +28,12 @@ function setStatementTimeout(client) {
...
 }
-function stringifyProtobuf(layers, tile) {
+
+// allowing tho options object for addGeoJson() - simplification settings, etc.
+function stringifyProtobuf(layers, tile, mvtOptions) {
   var vtile = new mapnik.VectorTile(tile.z, tile.x, tile.y);
   for (var layerName in layers) {
-    vtile.addGeoJSON(JSON.stringify(layers[layerName]), layerName);
-
+    vtile.addGeoJSON(JSON.stringify(layers[layerName]), layerName, mvtOptions);
```
Comment (abenrob) [inline lib/index.js]: the [addGeoJSON](http://mapnik.org/documentation/node-mapnik/3.5/#VectorTile.addGeoJSON) method is where we can influence simplification. Or any other options for that matter.

**Comment by devTristan** [comment]
I would rewrite this:

```js
  var mvtOptions = {};
  // if options were passed, we'll have 2 left
  if (callbacks.length === 2){
    // assign mvtOptions
    mvtOptions = arguments['1'];
    // shave off options
    callbacks.shift();
    // rewrite arguments
    arguments['1'] = arguments['2'];
    delete arguments['2'];
  }
```

more like this:

```js
var mvtOptions = {};

// if the last argument is an object, use it as mvtOptions
if (typeof callbacks[callbacks.length - 1] === 'object') {
  mvtOptions = callbacks.pop()
}

**Comment by abenrob** [comment]
@devTristan nice suggestion. Mine was breaking with middleware inclusion; it works now. Please review.

**Pull request merged**